### PR TITLE
fix(table): auto-collapse expanded row upon clicking another expandable row

### PR DESCRIPTION
**Pull request by ckrook**
## **Describe pull-request**  
This pull-request adds the possibility for the developer to enable "auto collapse". This means that if its enabled, all other rows except for the row selected will be closed.

## **Issue Linking:**  
_Choose one of the following options_
- **Jira:** [CDEP-3490](https://tegel.atlassian.net/browse/CDEP-3490)
- **GitHub:**  [Issue 379](https://github.com/scania-digital-design-system/tegel/issues/739)

## **How to test**  
_Provide detailed steps for testing, including any necessary setup._
1. Go to Storybook
2. Navigate to the table component 
3. Go to expandable rows
4. Toggle the control autoCollapse to true
5. Test open multiple rows now, the expected behavior is that all other rows expect the selected row should close

## **Checklist before submission**
- [x] I have added unit tests for my changes (if applicable)
- [x] All existing tests pass
- [x] I have updated the documentation (if applicable)

## **Suggested test steps**
- [x] Browser testing (Chrome, Safari, Firefox) 
- [x] Keyboard operability
- [x] Storybook controls

[CDEP-3490]: https://tegel.atlassian.net/browse/CDEP-3490?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by theJohnnyMe — APPROVED** [review]
Nice work @ckrook ! Looks good to me.
Please, correct title of PR according to conventional commits strategy ;)

File: packages/core/src/components/table/table-body-row-expandable/table-body-row-expandable.tsx [hunk 1]
```diff
@@ -101,6 +104,14 @@ export class TdsTableBodyRowExpandable {
     }
   }
 
+  @Listen('tdsChange', { target: 'body' })
+  handleRowExpand(event: CustomEvent<{ rowId: string; isExpanded: boolean }>) {
+    /** Collapse all other rows when autoCollapse is true and a row is expanded */
+    if (this.autoCollapse && event.detail.isExpanded && event.detail.rowId !== this.rowId) {
```
Comment (mJarsater) [inline packages/core/src/components/table/table-body-row-expandable/table-body-row-expandable.tsx:110]: Might wanna check that the tableId is the same as `this.tableId` aswell. Given that a page could have multiple tables and in theory the id of the rows should be shared between tables. For example if a user uses index as the id.

Comment (ckrook) [reply packages/core/src/components/table/table-body-row-expandable/table-body-row-expandable.tsx:110]: Good point which I did not think of, thanks!

**Comment by Lunkan89** [comment]
Nice work!

**Review by theJohnnyMe — APPROVED** [review]
Nice work @ckrook !

**Comment by mistermalm** [comment]
Not sure if we need the console.log in storybook? Otherwise looks good to me and works well to! Also great that you stuck with snapshots 🚀

**Review by mistermalm — APPROVED** [review]

**Review by nathalielindqvist — COMMENTED** [review]
There seems to be a lot of files in this PR, was it effected by a rebase in some way?

**Comment by ckrook** [comment]
Closing this issue because there seems to be an issue with the amount of files changed in this PR.
I have opened a new PR here:
https://github.com/scania-digital-design-system/tegel/pull/793

**Pull request closed without merging**